Python/pywarpx/picmi.py

```diff
@@ -2028,6 +2041,38 @@ def initialize_inputs(self):
             fields_to_plot.sort()
             self.diagnostic.fields_to_plot = fields_to_plot
 
+        particle_fields_to_plot = set()
+        if self.particle_fields_to_plot is not None:
+            for dataname, params in self.particle_fields_to_plot.items():
+                particle_fields_to_plot.add(dataname)
```

Comment:
Perhaps check to make sure that a data name is not repeated, since that is presumably an error.
```
if dataname in particle_fields_to_plot:
    raise Exception('A particle fields data name can not be repeated')
```
Note that particle_fields_to_plot can be a list (instead of a set).

Reply:
This is avoided by the fact that the `dataname` is a key in a dictionary, so duplicate values are not an issue